src/Umbraco.Web.Website/Controllers/UmbLoginController.cs

```diff
@@ -33,12 +51,133 @@ public class UmbLoginController : SurfaceController
+        ITwoFactorLoginService twoFactorLoginService,
+        IEmailSender emailSender,
+        IHttpContextAccessor httpContextAccessor,
+        LinkGenerator linkGenerator,
+        IOptions<WebRoutingSettings> webRoutingSettings,
+        IOptions<GlobalSettings> globalSettings)
         : base(umbracoContextAccessor, databaseFactory, services, appCaches, profilingLogger, publishedUrlProvider)
     {
         _signInManager = signInManager;
         _memberManager = memberManager;
         _twoFactorLoginService = twoFactorLoginService;
+        _emailSender = emailSender;
+        _httpContextAccessor = httpContextAccessor;
+        _linkGenerator = linkGenerator;
+        _webRoutingSettings = webRoutingSettings.Value;
+        _globalSettings = globalSettings.Value;
+    }
+
+    [Obsolete("Use ctor with all params")]
+    public UmbLoginController(
+        IUmbracoContextAccessor umbracoContextAccessor,
+        IUmbracoDatabaseFactory databaseFactory,
+        ServiceContext services,
+        AppCaches appCaches,
+        IProfilingLogger profilingLogger,
+        IPublishedUrlProvider publishedUrlProvider,
+        IMemberSignInManager signInManager)
+        : this(
+            umbracoContextAccessor,
+            databaseFactory,
+            services,
+            appCaches,
+            profilingLogger,
+            publishedUrlProvider,
+            signInManager,
+            StaticServiceProvider.Instance.GetRequiredService<IMemberManager>(),
+            StaticServiceProvider.Instance.GetRequiredService<ITwoFactorLoginService>(),
+            StaticServiceProvider.Instance.GetRequiredService<IEmailSender>(),
+            StaticServiceProvider.Instance.GetRequiredService<IHttpContextAccessor>(),
+            StaticServiceProvider.Instance.GetRequiredService<LinkGenerator>(),
+            StaticServiceProvider.Instance.GetRequiredService<IOptions<WebRoutingSettings>>(),
+            StaticServiceProvider.Instance.GetRequiredService<IOptions<GlobalSettings>>())
+    {
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    [ValidateUmbracoFormRouteString]
+    public async Task<IActionResult> HandleForgottenPassword([Bind(Prefix = "forgottenPasswordModel")] ForgottenPasswordModel model)
+    {
+        if (ModelState.IsValid == false)
+        {
+            return CurrentUmbracoPage();
+        }
+
+        MergeRouteValuesToModel(model);
+
+        if (!_emailSender.CanSendRequiredEmail())
+        {
+            return CurrentUmbracoPage();
+        }
+
+        MemberIdentityUser? memberIdentity = await _memberManager.FindByNameAsync(model.Username);
+        if (memberIdentity == null!)
+        {
+            return new ValidationErrorResult(
+                $"No local member found for username {model.Username}");
+        }
+
+        // send the email
+        await SendMemberResetEmailAsync(memberIdentity, null, null, null);
+
+        TempData["ForgottenPasswordSuccess"] = true;
```

Comment:
This isn't used anywhere, also as noted we should indicate neither success nor failure after submitting the form so as not to reveal if the account exists.